source/github-helpers/get-current-git-ref.ts

```diff
@@ -1,10 +1,30 @@
-import {getCurrentBranchFromFeed} from './index.js';
+import {isRepoCommitList} from 'github-url-detection';
+import select from 'select-dom';
+
+import {branchSelector} from './selectors.js';
 
 const typesWithGitRef = new Set(['tree', 'blob', 'blame', 'edit', 'commit', 'commits', 'compare']);
 const titleWithGitRef = / at (?<branch>[.\w-/]+)( · [\w-]+\/[\w-]+)?$/i;
 
-/** This only works with URL and page title. Must not be async because it's used by GitHubURL */
-export default function getCurrentGitRef(pathname = location.pathname, title = document.title): string | undefined {
+/** Must not be async because it's used by GitHubURL. May return different results depending on whether it's called before or after DOM ready */
+export default function getCurrentGitRef(): string | undefined {
+	// Note: This is not in the <head> so it's only available on AJAXed loads.
+	// It appears on every Code page except `commits` on folders/files
+	const refViaPicker = select(branchSelector)?.textContent!.trim();
+	if (refViaPicker) {
+		return refViaPicker;
+	}
+
+	// Slashed branches on `commits`, including pages without a branch picker
+	const branchFromFeed = getCurrentBranchFromFeed();
+	if (branchFromFeed) {
+		return branchFromFeed;
+	}
```

Comment:
I wanted to drop this, but decided to keep it because there's no branch selector on some commit pages, so this is the only way to get the current branch.